src/zimscraperlib/zim/creator.py

```diff
@@ -107,54 +105,72 @@ def __init__(
         self.workaround_nocancel = workaround_nocancel
         self.ignore_duplicates = ignore_duplicates
 
+    def set_mainpath(self, main_path):
+        super().set_mainpath(main_path)
+
     def start(self):
         super().__enter__()
 
-        if getattr(self, "main_path", None):
-            self.set_mainpath(self.main_path)
-
-        if not self.__all_metadata_is_set:
+        if not self._all_metadata_is_set:
             raise ValueError("Mandatory metadata are not all set.")
-        self.__add_metadata(**self.metadata)
+        self.add_metadata(**self._metadata)
+        self.add_default_illustration(self._metadata["Illustration_48x48@1"])
```

Comment:
Since we're specifing the size in our variable, we should use `add_illustration()`. Maybe we don't need `add_default_illustration` anymore then

Reply:
`add_default_illustration(content)` is just a wrapper of `add_illustration(48, content)`. 
so since we specified the illustration is 48x48, I think it’s fine here.

Reply:
It's purpose was just to ease with adding this llustration without having to bother with the name and size. Now that this metadata is mandatory and that we have to type `Illustration_48x48_at_1` anyway, it serves no purpose anymore. It's a good opportunity to remove it.